obspy/io/shapefile/core.py

```diff
@@ -0,0 +1,282 @@
+        # empty one with `None` values in it
+        origin = (event.preferred_origin() or
+                  event.origins and event.origins[0] or
+                  Origin(force_resource_id=False))
+        magnitude = (event.preferred_magnitude() or
+                     event.magnitudes and event.magnitudes[0] or
+                     Magnitude(force_resource_id=False))
+        t_origin = origin.time
+        pick_times = [pick.time for pick in event.picks
+                      if pick.time is not None]
+        t_pick = pick_times and min(pick_times) or None
+        date = t_origin or t_pick
+
+        feature = ogr.Feature(layer_definition)
+
+        # setting fields with `None` results in values of `0.000`
+        # need to really omit setting values if they are `None`
+        if event.resource_id is not None:
+            feature.SetField(native_str("EventID"),
+                             native_str(event.resource_id))
+        if origin.resource_id is not None:
+            feature.SetField(native_str("OriginID"),
+                             native_str(origin.resource_id))
+        if t_origin is not None:
+            # Use timestamp for exact timing
+            feature.SetField(native_str("OriginTime"), t_origin.timestamp)
+        if t_pick is not None:
+            # Use timestamp for exact timing
+            feature.SetField(native_str("FirstPick"), t_pick.timestamp)
+        if date is not None:
+            # ESRI shapefile attributes are stored in dbf files, which can not
+            # store datetimes, only dates.
+            # We still need to use the GDAL API with precision up to seconds
+            # (aiming at other output drivers of GDAL; `100` stands for GMT)
+            feature.SetField(native_str("Date"), date.year, date.month,
+                             date.day, date.hour, date.minute, date.second,
+                             100)
+        if origin.latitude is not None:
+            feature.SetField(native_str("Latitude"), origin.latitude)
+        if origin.longitude is not None:
+            feature.SetField(native_str("Longitude"), origin.longitude)
+        if origin.depth is not None:
+            feature.SetField(native_str("Depth"), origin.depth / 1e3)
+        if magnitude.mag is not None:
+            feature.SetField(native_str("Magnitude"), magnitude.mag)
+        if magnitude.resource_id is not None:
+            feature.SetField(native_str("Magnitu_ID"),
+                             native_str(magnitude.resource_id))
+
+        if origin.latitude is not None and origin.longitude is not None:
+            point = ogr.Geometry(ogr.wkbPoint)
+            point.AddPoint(origin.longitude, origin.latitude)
+            feature.SetGeometry(point)
+
+        layer.CreateFeature(feature)
+        # Destroy the feature to free resources
+        feature.Destroy()
+
+    # Destroy the data source to free resources
+    data_source.Destroy()
```

Comment:
Ah, I see it's destroyed here, okay. I think it makes more sense to destroy in `_write_shapefile`, though.